parboiled-core/src/main/scala/org/parboiled2/Parser.scala

```diff
@@ -473,6 +473,7 @@ abstract class Parser(initialValueStackSize: Int = 16, maxValueStackSize: Int =
     def length: Int                                       = input.length - offset
     def charAt(ix: Int): Char                             = input.charAt(offset + ix)
   }
+  def __subParserInput = new __SubParserInput()
```

Comment:
Calling `new __SubParserInput()` inside the scala 3 macro didn't seem to work. The existing scala 2 macro implementation has a little todo `// TODO: avoid re-allocation by re-using a cached instance` -- it occurred to me it would be fairly trivial to implement this def as
```
  private var __cachedSubParserInput: __SubParserInput = _
  def __subParserInput: __SubParserInput = {
    if (__cachedSubParserInput eq null) __cachedSubParserInput = new __SubParserInput()
    __cachedSubParserInput.resetOffset(_cursor)
    __cachedSubParserInput
  }
  ```
with a corresponding change to the __SubParserInput class of replacing `val offset = _cursor` with
```
    private var _offset: Int                              = 0
    private[Parser] def resetOffset(o: Int): Unit         = _offset = o
    def offset: Int                                       = _offset
```
  but I didn't do that here because that seems a bit out-of-scope and honestly I don't know if it would produce any improvement or how you guys profile the project.